### PR TITLE
Fix #6209 - introduce standalone dir crate

### LINKED ISSUE #6209: parity/dir.rs should be a standalone crate
and it should be used by `ethstore` instead of `ethstore/src/dir/paths.rs`

### PR DESCRIPTION
**Pull request by nicolasochem**
Done so far:

* created the dir crate in util
* moved code from `ethstore/src/dir/paths.rs` to dir crate
* rename `dir` module in ethstore to `accounts_dir` to distinguish it
  from the dir crate

Is there a need to mutualize some code in the newly created
`util/dir/src/lib.rs` ? If so, I would need some guidance as of which
functions are to be mutualized.

### CODE REVIEW

File: parity/rpc.rs [hunk 1]
```diff
@@ -21,7 +21,8 @@ use std::collections::HashSet;
 
 use dapps;
 use dir::default_data_path;
-use helpers::{parity_ipc_path, replace_home};
+use helpers::parity_ipc_path;
```
Comment (tomusdrw) [inline parity/rpc.rs]: Could you keep the imports sorted?

File: util/dir/src/helpers.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/// Replaces `$HOME` str with home directory path.
+pub fn replace_home(base: &str, arg: &str) -> String {
+	// the $HOME directory on mac os should be `~/Library` or `~/Library/Application Support`
+	let r = arg.replace("$HOME", env::home_dir().unwrap().to_str().unwrap());
+	let r = r.replace("$BASE", base);
+	r.replace("/", &::std::path::MAIN_SEPARATOR.to_string())
+}
+
+pub fn replace_home_and_local(base: &str, local: &str, arg: &str) -> String {
```
Comment (tomusdrw) [inline util/dir/src/helpers.rs]: Doc missing.

File: util/dir/src/lib.rs [hunk 3]
```diff
@@ -14,7 +14,12 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::fs;
+extern crate app_dirs;
```
Comment (tomusdrw) [inline util/dir/src/lib.rs]: Please add `#![warn(missing_docs)]`

File: util/dir/src/lib.rs [hunk 4]
```diff
@@ -234,6 +239,82 @@ pub fn default_hypervisor_path() -> String {
...
+}
+
+/// Parity path for specific chain
+pub fn parity(chain: &str) -> PathBuf {
+	let mut base = parity_base();
+	base.push(chain);
+	base
+}
+
+#[cfg(target_os = "macos")]
```
Comment (tomusdrw) [inline util/dir/src/lib.rs]: Might be better to group the functions into a `platform` module:
```
#[cfg(target_os = "macos")]
mod platform {
  fn parity_base() {}
  fn geth_base() {}
}
```

**Comment by tomusdrw** [comment]
@nicolasochem any progress?

**Comment by nicolasochem** [comment]
replaced by #7383

**Pull request closed without merging**